lib/adapters/xhr.js

```diff
@@ -142,14 +142,15 @@ module.exports = function xhrAdapter(config) {
     }
 
     // Handle progress if needed
-    if (typeof config.progress === 'function') {
-      if (config.method === 'post' || config.method === 'put') {
-        request.upload.addEventListener('progress', config.progress);
-      } else if (config.method === 'get') {
-        request.addEventListener('progress', config.progress);
-      }
+    if (typeof config.progressDownload === 'function') {
+      request.addEventListener('progress', config.progressDownload);
+    }
+
+    if (typeof config.progressUpload === 'function') {
+      request.upload.addEventListener('progress', config.progressUpload);
```

Comment:
It doesn't look like `upload` is supported by all modern browsers (please correct me if I'm wrong) so I would suggest to check whether this property is present before calling `addEventListener`: 

```js
if (typeof config.progressUpload === 'function' && request.upload) {
  // ...
}
```